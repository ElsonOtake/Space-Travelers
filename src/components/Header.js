import React from 'react';
import Container from 'react-bootstrap/Container';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';

const Header = () => (
  <Navbar bg="light" variant="light">
    <Container>
      <Navbar.Brand href="/">Space Travelers' Hub</Navbar.Brand>
      <Nav>
        <Nav.Link 
          href="/"
          className={({ isActive }) => (isActive ? 'active' : 'notActive')}
        >
          Rockets
        </Nav.Link>
        <Nav.Link 
          href="/missions"
          className={({ isActive }) => (isActive ? 'active' : 'notActive')}
        >
          Missions
        </Nav.Link>
        <Nav.Link 
          href="/my_profile"
          className={({ isActive }) => (isActive ? 'active' : 'notActive')}
        >
          My Profile
        </Nav.Link>
      </Nav>
    </Container>
  </Navbar>
);

export default Header;