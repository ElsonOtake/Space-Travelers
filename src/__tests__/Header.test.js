import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/Header';

describe('Tests for the Header component', () => {
  render(
    <Router>
      <Header />
    </Router>,
  );
  test('Checks for the following tests on screen', () => {
    expect(screen.getByText(/Space Travelers/)).toBeTruthy();
    expect(screen.getByText(/Hub/)).toBeTruthy();
    expect(screen.getByText("Space Travelers' Hub")).toBeTruthy();
    expect(screen.getByText('Rockets')).toBeTruthy();
    expect(screen.getByText('Missions')).toBeTruthy();
    expect(screen.getByText('My Profile')).toBeTruthy();

    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();

    expect(screen.getAllByRole('link').length).toBe(3);

    expect(screen.getByAltText('planet logo')).toBeTruthy();
  });
  test('Checks for the Snapshot', () => {
    expect(screen.debug()).toMatchSnapshot();
  });
});
