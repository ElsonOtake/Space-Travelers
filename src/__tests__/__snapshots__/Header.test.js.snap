// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Header component Checks for the Snapshot 1`] = `undefined`;
